File: mongodbatlas/user/encrypt.js
#!/usr/bin/env node

'use strict';

var uri = process.argv[2];
var key     = process.argv[3];
var Crypto = require("crypto")

const IV_LENGTH = 16;
    
function encrypturi(message, salt) {
    if (salt.length !== 32) {
        throw new Error(`Encryption Key length MUST be 32 characters!`);
    }
    let iv = Crypto.randomBytes(IV_LENGTH);
    let cipher = Crypto.createCipheriv("aes-256-cbc", Buffer.from(salt),iv);
    let encrypted = cipher.update(message);
    encrypted = Buffer.concat([encrypted, cipher.final()]);
    const result = iv.toString("hex") + ":" + encrypted.toString("hex");
    return result;
}

console.log( encrypturi(uri,key));

File: mongodbatlas/user/encrypt.py
from Crypto.Cipher import AES
from Crypto.Util.Padding import pad
from sys import argv

scriptname, uri, key, iv = argv

def encryptUri(uri, key, iv):
    if (len(key) != 32):
        raise Exception("Sorry, encryption key must be 32 characters long.")
    uri_bytes=uri.encode('utf-8')
    key_bytes=key.encode('utf-8')
    iv_bytes=iv.encode('utf-8')
    cipher = AES.new(key_bytes, AES.MODE_CBC, iv=iv_bytes)
    ct_bytes = cipher.encrypt(pad(uri_bytes, AES.block_size))
    iv = cipher.iv.hex()
    ct = ct_bytes.hex()
    result = f"{iv}:{ct}"
    return result

def main():
    print(encryptUri(uri, key, iv))

if __name__ == "__main__":
    main()